lib/partitions.js

```diff
@@ -130,6 +131,19 @@ Partition.prototype.produce = function() {
     }.bind(this));
 }
 
+/**
+ * Get a Partitions resource representing the partitions for this topic. This does not request any metadata since you may
+ * use the Partitions resource to get a specific partition; call list() on the resulting Partitions resource to get a
+ * full listing with metadata.
+ *
+ * @param res function(err,
+ */
+Partition.prototype.__defineGetter__("messages", function() {
+    if (!this._messages)
+        this._messages = new PartitionMessages(this);
+    return this._messages.messages.bind(this._messages);
```

Comment:
Initially, I did it this way is because i felt it awkward to write code like this.

`kafka.topics.topic('test').partitions().partition(0).messages().messages({offset: 0}, callback);`

Nothing necessarily wrong with that and yes, it allows us to add more actions in the future if required. I justified my decision by looking at the url structure in kafka rest. Each url segment is directly related to a method in `kafka-rest-node`

`/topics/test/partitions/0/messages?offset=0`
`topics.topic('test').partitions().partition(0).messages({offset: 0}, callback)`

To me, that feels nicer but either works.